File: sdk/golang/pkg/generate/service/uta_api.go
package service

import (
	"github.com/Kucoin/kucoin-universal-sdk/sdk/golang/internal/interfaces"
	"github.com/Kucoin/kucoin-universal-sdk/sdk/golang/pkg/generate/uta/account"
	"github.com/Kucoin/kucoin-universal-sdk/sdk/golang/pkg/generate/uta/affiliate"
	"github.com/Kucoin/kucoin-universal-sdk/sdk/golang/pkg/generate/uta/market"
	"github.com/Kucoin/kucoin-universal-sdk/sdk/golang/pkg/generate/uta/order"
	"github.com/Kucoin/kucoin-universal-sdk/sdk/golang/pkg/generate/uta/positions"
	"github.com/Kucoin/kucoin-universal-sdk/sdk/golang/pkg/generate/uta/viplending"
)

type UTAService interface {
	GetAccountAPI() account.AccountAPI
	GetAffiliateAPI() affiliate.AffiliateAPI
	GetMarketAPI() market.MarketAPI
	GetOrderAPI() order.OrderAPI
	GetPositionsAPI() positions.PositionsAPI
	GetVIPLendingAPI() viplending.VIPLendingAPI
}

type UTAServiceImpl struct {
	AccountAPI    account.AccountAPI
	AffiliateAPI  affiliate.AffiliateAPI
	MarketAPI     market.MarketAPI
	OrderAPI      order.OrderAPI
	PositionsAPI  positions.PositionsAPI
	VIPLendingAPI viplending.VIPLendingAPI
}

func NewUTAService(transport interfaces.Transport) UTAService {
	api := &UTAServiceImpl{}
	api.AccountAPI = account.NewAccountAPIImp(transport)
	api.AffiliateAPI = affiliate.NewAffiliateAPIImp(transport)
	api.MarketAPI = market.NewMarketAPIImp(transport)
	api.OrderAPI = order.NewOrderAPIImp(transport)
	api.PositionsAPI = positions.NewPositionsAPIImp(transport)
	api.VIPLendingAPI = viplending.NewVIPLendingAPIImp(transport)
	return api
}

func (impl *UTAServiceImpl) GetAccountAPI() account.AccountAPI {
	return impl.AccountAPI
}

func (impl *UTAServiceImpl) GetAffiliateAPI() affiliate.AffiliateAPI {
	return impl.AffiliateAPI
}

func (impl *UTAServiceImpl) GetMarketAPI() market.MarketAPI {
	return impl.MarketAPI
}

func (impl *UTAServiceImpl) GetOrderAPI() order.OrderAPI {
	return impl.OrderAPI
}

func (impl *UTAServiceImpl) GetPositionsAPI() positions.PositionsAPI {
	return impl.PositionsAPI
}

func (impl *UTAServiceImpl) GetVIPLendingAPI() viplending.VIPLendingAPI {
	return impl.VIPLendingAPI
}

File: sdk/node/src/generate/service/uta_api.ts

import { Transport } from '@internal/interfaces/transport';

import { AccountAPI, AccountAPIImpl } from '@generate/uta/account/api_account';
import { AffiliateAPI, AffiliateAPIImpl } from '@generate/uta/affiliate/api_affiliate';
import { MarketAPI, MarketAPIImpl } from '@generate/uta/market/api_market';
import { OrderAPI, OrderAPIImpl } from '@generate/uta/order/api_order';
import { PositionsAPI, PositionsAPIImpl } from '@generate/uta/positions/api_positions';
import { VIPLendingAPI, VIPLendingAPIImpl } from '@generate/uta/viplending/api_vip_lending';

export interface UTAService {
    getAccountApi(): AccountAPI;
    getAffiliateApi(): AffiliateAPI;
    getMarketApi(): MarketAPI;
    getOrderApi(): OrderAPI;
    getPositionsApi(): PositionsAPI;
    getVIPLendingApi(): VIPLendingAPI;
}

export class UTAServiceImpl implements UTAService {
    private readonly transport: Transport;
    private readonly account: AccountAPI;
    private readonly affiliate: AffiliateAPI;
    private readonly market: MarketAPI;
    private readonly order: OrderAPI;
    private readonly positions: PositionsAPI;
    private readonly vipLending: VIPLendingAPI;

    constructor(transport: Transport) {
        this.transport = transport;
        this.account = new AccountAPIImpl(transport);
        this.affiliate = new AffiliateAPIImpl(transport);
        this.market = new MarketAPIImpl(transport);
        this.order = new OrderAPIImpl(transport);
        this.positions = new PositionsAPIImpl(transport);
        this.vipLending = new VIPLendingAPIImpl(transport);
    }

    getAccountApi(): AccountAPI {
        return this.account;
    }

    getAffiliateApi(): AffiliateAPI {
        return this.affiliate;
    }

    getMarketApi(): MarketAPI {
        return this.market;
    }

    getOrderApi(): OrderAPI {
        return this.order;
    }

    getPositionsApi(): PositionsAPI {
        return this.positions;
    }

    getVIPLendingApi(): VIPLendingAPI {
        return this.vipLending;
    }
}
